node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/MetricsReporter.java

```diff
@@ -54,7 +55,10 @@
 
     @Override
     public void maintain() {
-        List<Node> nodes = nodeRepository().getNodes();
+        LockedNodeList nodes;
+        try (var lock = nodeRepository().lockAllocation()) {
+            nodes = nodeRepository().list(lock);
```

Comment:
Looks like this is done only because `updateDockerMetrics()` is using `DockerHostCapacity`, but otherwise this could be the regular `list()` method. Maybe just use a fake/no-op mutex like `() -> {}` so its clear that mutex is not important here?

Ideally the methods that depend on the mutex and take in the `LockedNodeList` should somehow verify that the mutex has not been closed yet...